src/Hazelcast.Net/Serialization/Compact/CompactSerializationSerializer.cs

```diff
@@ -0,0 +1,375 @@
+                _registrationsById.TryAdd(schema.Id, registration);
+
+                // now that we have a new schema, we need to publish it, unless specified
+                // otherwise by the registration.
+                _schemas.Add(schema, registration.IsClusterSchema);
+            }
+
+            WriteSchema(output, schema);
+            var writer = new CompactWriter(this, output, schema);
+            registration.Serializer.Write(writer, obj);
+            writer.Complete();
+        }
+
+        private static void WriteSchema(ObjectDataOutput output, Schema schema)
+        {
+            output.WriteLong(schema.Id);
+
+            // note: this is the code we would use if we were to send the schema alongside
+            // the data, and we keep it here for reference only, just in case one day...
+
+            /*
+            if (!withSchema) return;
+
+            var startPosition = output.Position;
+            output.WriteInt(0);
+            var schemaPosition = output.Position;
+
+            // note: don't output.WriteObject(schema) else it's serialized as identified serializable
+            output.WriteString(schema.TypeName);
+            output.WriteInt(schema.Fields.Count);
+            foreach (var field in schema.Fields)
+            {
+                output.WriteString(field.FieldName);
+                output.WriteInt((int)field.Kind);
+            }
+
+            var position = output.Position;
+            output.MoveTo(startPosition);
+            output.WriteInt(position - schemaPosition);
+            output.MoveTo(position);
+            */
+        }
+
+        private static Schema BuildSchema(CompactRegistration registration, object obj)
+        {
+            var builder = new SchemaBuilderWriter(registration.TypeName);
+            registration.Serializer.Write(builder, obj);
+            return builder.Build();
+        }
+
+        public bool TryRead(IObjectDataInput input, out object? obj, out long missingSchemaId)
+        {
+            if (!(input is ObjectDataInput inputInstance))
+                throw new ArgumentException("Input must be an ObjectDataInput instance.", nameof(input));
+
+            return TryReadObject(inputInstance, out obj, out missingSchemaId);
+        }
+
+        private bool TryReadObject(ObjectDataInput input, out object? obj, out long missingSchemaId)
+        {
+            var schemaId = input.ReadLong();
+            if (!_schemas.TryGet(schemaId, out var schema))
+            {
+                obj = null;
+                missingSchemaId = schemaId;
+                return false;
+            }
+
+            var registration = GetOrCreateRegistration(schema);
+
+            var reader = new CompactReader(this, input, schema, registration.SerializedType);
+            // FIXME - but then, each serializer MUST support a TryRead method AND return a state and oh my...
```

Comment:
So, there is no `TryXXX()` approach in other clients as wwll :|

Reply:
True. And that was a leftover comment, thanks for catching it, removing it. And in fact the problem it mentioned is solved in a different way so it's totally not relevant anyways.